yang/vyatta-protocols-frr-zebra-v1.yang

```diff
@@ -0,0 +1,227 @@
+  contact
+    "Niral Networks
+    Postal: Sigma Soft Tech Park, Whitefield Main Road,
+    Bengaluru, India 560066
+    Web: www.niralnetworks.com";
+
+  description
+    "Copyright (c) 2019-2020, Niral Networks Intellectual Property.
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or
+    without modification, are permitted provided that the following
+    conditions are met:
+
+    1. Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above
+    copyright notice, this list of conditions and the following
+    disclaimer in the documentation and/or other materials provided
+    with the distribution.
+    3. Neither the name of the copyright holder nor the names of its
+    contributors may be used to endorse or promote products derived
+    from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    'AS IS' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+
+    SPDX-License-Identifier: BSD-3-Clause
+    The YANG module package for vyatta-protocols-frr-zebra-v1";
+
+  revision 2021-03-02 {
+    description 
+      "Added link-params option.";
+  }
+  
+	typedef ieee-float {
+  	type string {
+    	pattern
+      	'0[xX](0((\.0?)?[pP](\+)?0?|(\.0?))|'
+    	+ '1(\.([0-9a-fA-F]{0,5}[02468aAcCeE]?)?)?[pP](\+)?(12[0-7]|'
+    	+ '1[01][0-9]|0?[0-9]?[0-9])?)';
+  	}
+    configd:help "IEEE floating point format";
+	}
+	
+	typedef float {
+  	type string {
+    	pattern
+      	'(0((\.0?)?[pP](\+)?0?|(\.0?))|'
+    	+ '1(\.([0-9a-fA-F]{0,5}[02468aAcCeE]?)?)?[pP](\+)?(12[0-7]|'
+    	+ '1[01][0-9]|0?[0-9]?[0-9])?)';
+  	}
+    configd:help "floating point format";
+	}
+
+
+  grouping link-params {
+    container link-params {
+      configd:help "Configure interface link parameters";
+			leaf admin-grp {
+				type string {
+      		pattern "0[Xx][a-fA-F0-9]+";
+    		}
+				must "(../enable)" {
+					error-message "link-params enable must be set.";
+				}
+      	configd:help "Administrative group membership";
+				description "Administrative group membership";
+			}
+			leaf enable {
```

Comment:
Why is there an explicit enable here? What does this add?

Reply:
The explicit enable has been given as per the functionality stated in FRR i.e for activating link parameters on the interface

Reply:
The model should model the concept well, not the underlying implementation. Are users likely to want to leave the parameters configured without them taking effect? My guess would be probably not, but I'm opening to hear if you have other options on that.

My preference would be dropping this entirely, but if there is use in users being able to disable the parameters without unconfiguring them, this should be done in a similar way to firewall rules where there is a disable rather than an enable (or an enum of enabled/disabled with a default of enabled).

Reply:
Yes, I agree. I have dropped the explicit enable in the latest commit.